cli/cppcheckexecutor.cpp

```diff
@@ -158,7 +158,33 @@ bool CppCheckExecutor::parseFromArgs(CppCheck *cppcheck, int argc, const char* c
 #else
     const bool caseSensitive = true;
 #endif
-    if (!pathnames.empty()) {
+    if (!mSettings->project.fileSettings.empty()) {
+        // filter only for the selected filenames from all project files
+        std::list<ImportProject::FileSettings> newList;
+
+        for (const std::string fname : pathnames) {
```

Comment:
I don't believe it's necessary to create copies of pathnames items so please make fname a reference.